itest/rust/src/register_tests/func_test.rs

```diff
@@ -46,3 +46,27 @@ impl RefCountedVirtual for FuncRename {
         Self
     }
 }
+
+#[derive(GodotClass)]
+#[class(base=RefCounted)]
+struct GdSelfReference;
+
+#[godot_api]
+impl GdSelfReference {
+    #[func(gd_self)]
+    fn takes_gd_as_self(this: Gd<GdSelfReference>) -> bool {
+        true
+    }
+
+    #[func(gd_self)]
+    fn takes_gd_as_self_mut(mut this: Gd<GdSelfReference>) -> bool {
+        true
+    }
+}
+
+#[godot_api]
+impl RefCountedVirtual for GdSelfReference {
+    fn init(_base: Base<Self::Base>) -> Self {
+        Self
+    }
```

Comment:
This could be replaced by `#[class(init)]` at the struct declaration.

The empty `impl` may still be necessary due to a bug.

Reply:
Seems to work.